### PR TITLE
Update jaeger operator to support strimzi operator 0.23.0

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

## Which problem is this PR solving?
- This PR adds compatibility with strimzi operator 0.23

## Short description of the changes
- Migrated kafka CRD to `v1beta2` API version

### CODE REVIEW

**Review by jpkrohling — COMMENTED** [review]
I think we talked about expanding the compatibility matrix in the readme to mention which Strimzi versions are supported for each release. This change isn't part of this PR.

**Comment by rubenvp8510** [comment]
> I think we talked about expanding the compatibility matrix in the readme to mention which Strimzi versions are supported for each release. This change isn't part of this PR.

Not yet, but I'll  add the matrix to this PR ;)

**Comment by rubenvp8510** [comment]
I just added the matrix to the `README`.

File: README.md [hunk 1]
```diff
@@ -71,6 +71,17 @@ The Jaeger Operator *might* work on versions outside of the given range, but whe
 | v1.23           | v1.19+               |
 | v1.22           | v1.18 to v1.20       |
 
+### Jaeger Operator vs. Strimzi Operator
```
Comment (jpkrohling) [inline README.md:74]: It might be better to use the table above, so that we update only one place when we need to perform a release.

Comment (rubenvp8510) [reply README.md:74]: Done!

File: README.md [hunk 2]
```diff
@@ -66,10 +66,17 @@ For instance, when we released v1.22.0, the latest Kubernetes version was v1.20.
 
 The Jaeger Operator *might* work on versions outside of the given range, but when opening new issues, please make sure to test your scenario on a supported version.
 
-| Jaeger Operator | Kubernetes           |
-|-----------------|----------------------|
-| v1.23           | v1.19+               |
-| v1.22           | v1.18 to v1.20       |
+| Jaeger Operator | Kubernetes           | Strimzi Operator   |
```
Comment (jpkrohling) [inline README.md:69]: Reading the doc in a broader context, I believe this table should be moved one section up, as it includes versions from two sections.

**Comment by rafilkmp3** [comment]
Please merge this,  latest version its failing to deploy since operator droped suport to v1beta2 

https://strimzi.io/docs/operators/0.24.0/deploying.html#assembly-upgrade-resources-str

image

File: README.md [hunk 3]
```diff
@@ -66,10 +75,13 @@ For instance, when we released v1.22.0, the latest Kubernetes version was v1.20.
...
 The Jaeger Operator *might* work on versions outside of the given range, but when opening new issues, please make sure to test your scenario on a supported version.
 
-| Jaeger Operator | Kubernetes           |
-|-----------------|----------------------|
-| v1.23           | v1.19+               |
-| v1.22           | v1.18 to v1.20       |
+
+### Jaeger Operator vs. Strimzi Operator
+
+We maintain compatibility with a set of tested Strimzi operator versions, but some changes in Strimzi operator require us to break compatibility with older versions
```
Comment (jpkrohling) [inline README.md:81]:
```suggestion
We maintain compatibility with a set of tested Strimzi operator versions, but some changes in Strimzi operator require us to break compatibility with older versions.
```

**Review by jpkrohling — APPROVED** [review]
I'll approve as is but will send another PR with a suggestion for the readme.

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Could you please fix the conflicts?

**Pull request merged**